app/javascript/components/subject/CaseStatus.js

```diff
@@ -29,37 +29,42 @@ class CaseStatus extends React.Component {
     event.persist();
     const value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
 
-    this.setState({ [event.target.id]: value }, () => {
-      if (event.target.id === 'case_status') {
-        const confirmedOrProbable = value === 'Confirmed' || value === 'Probable';
-        const hideModal = this.state.isolation && confirmedOrProbable && this.props.patient.case_status !== '';
+    if (event?.target?.name && event.target.name === 'apply_to_group') {
+      let applyToGroup = event.target.id === 'apply_to_group_yes';
+      this.setState({ [event.target.name]: applyToGroup });
+    } else {
+      this.setState({ [event.target.id]: value }, () => {
+        if (event.target.id === 'case_status') {
+          const confirmedOrProbable = value === 'Confirmed' || value === 'Probable';
+          const hideModal = this.state.isolation && confirmedOrProbable && this.props.patient.case_status !== '';
 
-        this.setState({ showCaseStatusModal: !hideModal, confirmedOrProbable }, () => {
-          // specific case where case status is just changed with no modal
-          if (hideModal) {
-            this.submit();
+          this.setState({ showCaseStatusModal: !hideModal, confirmedOrProbable }, () => {
+            // specific case where case status is just changed with no modal
+            if (hideModal) {
+              this.submit();
+            }
+            if (!confirmedOrProbable) {
```

Comment:
This should have the same check for a blank value as the other PR has. I can merge that one and then maybe you can update this one?

Reply:
yes I was planning on rebasing off whichever was merged first

Reply:
Just merged the other one.

Reply:
rebased